framework/src/outputs/XDA.C

```diff
@@ -44,15 +44,17 @@ XDA::XDA(const InputParameters & parameters) :
 void
 XDA::output(const ExecFlagType & /*type*/)
 {
+  std::string fname = filename();
+  std::string base = fname.substr(0, fname.size()-4);
   if (_binary)
   {
-    _mesh_ptr->getMesh().write(filename()+"_mesh.xdr");
-    _es_ptr->write (filename()+".xdr", ENCODE, EquationSystems::WRITE_DATA | EquationSystems::WRITE_ADDITIONAL_DATA);
+    _mesh_ptr->getMesh().write(base + "_mesh.xdr");
```

Comment:
You are effectively not using `filename()` at all now. Yes you call it to retrieve the filename with the suffix but then you immediately strip the suffix then manually add it back in here and just below? This seems clunky, I assume you need filename for an external interface but perhaps you should make use of the suffix you strip too?